lib/cosmos/script/scripting.rb

```diff
@@ -160,6 +160,29 @@ def check_raw(*args)
       _check(*args) {|tgt,pkt,item| tlm_raw(tgt,pkt,item) }
     end
 
+    # Executes the passed method and expects and exception to be raised.
+    # Raises a CheckError if an Exception is not raised.
+    # Usage:
+    #   check_exception { cmd(target_name, cmd_name, cmd_params = {}) }
+    # or
+    #   check_exception { cmd('target_name cmd_name with cmd_param1 value1, cmd_param2 value2') }
+    def check_exception(&block)
+      method = ''
+      begin
+        begin
+          # Try to get the block source. This doesn't always work ...
+          method = block.source.split('{')[1].split('}')[0].strip
```

Comment:
I don't know where the 'source' method is even implemented. Sometimes it throws a "Could not parse source for #<Proc:0x00000007a80988@C:/git/COSMOS/demo/procedures/procedure.rb:5>: unexpected $end" exception. I tried googling that and no one knows anything about it!